src/app/AppInitGate.tsx

```diff
@@ -1,61 +1,72 @@
+import { Dimensions } from 'react-native'
 import { findBestAvailableLanguage } from 'react-native-localize'
-import { useSelector } from 'react-redux'
+import { useDispatch, useSelector } from 'react-redux'
+import { AppEvents } from 'src/analytics/Events'
 import ValoraAnalytics from 'src/analytics/ValoraAnalytics'
-import { DEFAULT_APP_LANGUAGE } from 'src/config'
-import { initI18n } from 'src/i18n'
-import {
-  allowOtaTranslationsSelector,
-  currentLanguageSelector,
-  otaTranslationsAppVersionSelector,
-} from 'src/i18n/selectors'
+import { appMounted } from 'src/app/actions'
+import { useDeepLinks } from 'src/app/useDeepLinks'
+import i18n from 'src/i18n'
+import { currentLanguageSelector } from 'src/i18n/selectors'
 import useChangeLanguage from 'src/i18n/useChangeLanguage'
 import { navigateToError } from 'src/navigator/NavigationService'
+import { waitUntilSagasFinishLoading } from 'src/redux/sagas'
 import Logger from 'src/utils/Logger'
 
 const TAG = 'AppInitGate'
 
 interface Props {
-  loading: React.ReactNode
+  reactLoadTime: number
+  appStartedMillis: number
   children: React.ReactNode
 }
 
-const AppInitGate = ({ loading, children }: Props) => {
+const AppInitGate = ({ appStartedMillis, reactLoadTime, children }: Props) => {
   const changelanguage = useChangeLanguage()
-  const allowOtaTranslations = useSelector(allowOtaTranslationsSelector)
-  const otaTranslationsAppVersion = useSelector(otaTranslationsAppVersionSelector)
+  const dispatch = useDispatch()
+  const { addDeepLinkListeners } = useDeepLinks()
+
   const language = useSelector(currentLanguageSelector)
   const bestLanguage = findBestAvailableLanguage(Object.keys(locales))?.languageTag
-
-  const i18nInitializer = async () => {
-    await initI18n(
-      language || bestLanguage || DEFAULT_APP_LANGUAGE,
-      allowOtaTranslations,
-      otaTranslationsAppVersion
-    )
-    if (!language && bestLanguage) {
-      await changelanguage(bestLanguage)
-    }
-  }
-
   const initResult = useAsync(
     async () => {
-      Logger.debug(TAG, 'Starting init')
-      await Promise.all([i18nInitializer(), ValoraAnalytics.init()])
-      Logger.debug(TAG, 'init completed')
+      Logger.debug(TAG, 'Starting AppInitGate init')
+      await waitUntilSagasFinishLoading()
+
+      if (!language && bestLanguage) {
+        await changelanguage(bestLanguage)
+      }
+
+      const reactLoadDuration = (reactLoadTime - appStartedMillis) / 1000
+      const appLoadDuration = (Date.now() - appStartedMillis) / 1000
+      Logger.debug('TAG', `reactLoad: ${reactLoadDuration} appLoad: ${appLoadDuration}`)
+
+      const { width, height } = Dimensions.get('window')
+      ValoraAnalytics.startSession(AppEvents.app_launched, {
+        deviceHeight: height,
+        deviceWidth: width,
+        reactLoadDuration,
+        appLoadDuration,
+        language: i18n.language || language,
+      })
+
+      Logger.debug(TAG, 'AppInitGate init completed')
+      dispatch(appMounted())
```

Comment:
Now that I'm seeing this, I wonder if this part should maybe remain in `App.tsx` and called by a `onLoaded` callback passed to `AppInitGate`. The deeplinks would remain in `App.tsx` too.

Reply:
curious, what is the benefit of keeping this code in App.tsx? it'd be less changes to this PR of course, but aside from the reactLoadTime and appStartedMillis variables none of the other logic is tied to this component. we'd have functions here that are executed in a child component, why not put the function where it will be run?

Reply:
Yes I don't know, it's more a gut feeling, this part is arguably not directly part of the init.
It's what happens after init is done. Seeing it within `AppInitGate` feels like the component does too many things.
"oh look it also handles deeplinks".